giraffe/src/utils/legend/useTooltipStyle.ts

```diff
@@ -125,13 +132,23 @@ export const useAnnotationStyle = (
         }
       }
 
-      const clampedX = Math.round(Math.max(dx + x, x))
+      let clampedX = Math.round(x + dx)
       const clampedY = Math.round(dy + y)
 
+      // When the annotation is in the far edge of the screen, the position.left value
+      // overrides the width of the tooltip and makes its width smaller than its max-width.
+      // Fix the left value so that the tooltip is at its max width.
+      if (
+        window.innerWidth - clampedX < ANNOTATION_DEFAULT_MAX_WIDTH &&
+        tooltipWidth >= ANNOTATION_DEFAULT_MAX_WIDTH
```

Comment:
```&&
        tooltipWidth >= ANNOTATION_DEFAULT_MAX_WIDTH```

confused about this condition, what's it for?

Reply:
Without it, small tooltips (that have a width smaller than the max width) were being scooted to the side!